modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java

```diff
@@ -115,20 +112,43 @@ public String getObjectInstanceName() {
                     } if(!s.isBoundToTopic()){
                         continue;
                     }
-                    //Filter multiple shared subscriptions in disconnected mode. Because in UI
-                    // only one inactive shared subscription should be shown.
+
                     if (true == s.isDurable()) {
-                        if (uniqueueSubscriptionIDs.contains(s.getTargetQueue()) && !(s.hasExternalSubscriptions())) {
-                            continue;
+                        if (s.hasExternalSubscriptions()) {
+                            uniqueSubscriptionIDs.add(s.getTargetQueue());
                         } else {
-                            uniqueueSubscriptionIDs.add(s.getTargetQueue());
+                            // Since only one inactive shared subscription should be shown
+                            // we replace the existing value if any
+                            inactiveSubscriptions.put(s.getTargetQueue(), s);
+                            // Inactive subscriptions will be added later considering shared subscriptions
+                            continue;
                         }
                     }
 
-                    allSubscriptionsForTopics.add(renderSubscriptionForUI(s,pendingMessageCount.intValue()));
+                    subscriptionsToDisplay.add(s);
+                }
+
+                // In UI only one inactive shared subscription should be shown if there are no active subscriptions.
+                for (Map.Entry<String, AndesSubscription> inactiveEntry : inactiveSubscriptions.entrySet()) {
+                    // If there are active subscriptions with same target queue, we skip adding inactive subscriptions
+                    if (!uniqueSubscriptionIDs.contains(inactiveEntry.getKey())) {
```

Comment:
Change it like !(uniqueSubscriptionIDs.contains(inactiveEntry.getKey())) for more readability